File: api/db/test_db.py
import sys
import peewee
from .connection import db
from .db_models import Tenant, User, UserTenant

def main():
    db.connect()
    
    # Verify Go data exists
    t_go = Tenant.get(Tenant.id == 't-go')
    assert t_go.name == 'Go Tenant'
    
    # Insert Python data
    Tenant.create(id='t-py', name='Python Tenant')
    User.create(id='u-py', email='py@devrag.local', password_hash='hash')
    UserTenant.create(user_id='u-py', tenant_id='t-py', role='admin')
    
    # Test unique constraint on email
    try:
        User.create(id='u-py-2', email='py@devrag.local', password_hash='hash')
        sys.exit("Failed: Email unique constraint not enforced")
    except peewee.IntegrityError:
        pass
        
    # Test composite unique constraint on UserTenant
    try:
        UserTenant.create(user_id='u-py', tenant_id='t-py', role='invite')
        sys.exit("Failed: UserTenant unique constraint not enforced")
    except peewee.IntegrityError:
        pass

    print("Python reading, insertion, and constraints successful.")

if __name__ == '__main__':
    main()

File: cmd/test_db/test_db.go
package main

import (
	"fmt"
	"log"

	"github.com/6sLOGAN78/devRAG/internal/config"
	"github.com/6sLOGAN78/devRAG/internal/dao"
)

func main() {
	cfg, err := config.LoadConfig("conf/service_conf.yaml")
	if err != nil {
		log.Fatal(err)
	}

	err = dao.InitDB(cfg)
	if err != nil {
		log.Fatal(err)
	}

	// Auto Migrate
	err = dao.DB.AutoMigrate(&dao.Tenant{}, &dao.User{}, &dao.UserTenant{})
	if err != nil {
		log.Fatal("AutoMigrate failed: ", err)
	}

	// Insert test data
	tenant := dao.Tenant{ID: "t-go", Name: "Go Tenant"}
	dao.DB.Create(&tenant)

	user := dao.User{ID: "u-go", Email: "go@devrag.local", PasswordHash: "hash"}
	dao.DB.Create(&user)

	ut := dao.UserTenant{UserID: "u-go", TenantID: "t-go", Role: "owner"}
	dao.DB.Create(&ut)

	fmt.Println("Go migration and insertion successful.")
}
